staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -219,6 +219,10 @@ type Config struct {
 	// RequestWidthEstimator is used to estimate the "width" of the incoming request(s).
 	RequestWidthEstimator flowcontrolrequest.WidthEstimatorFunc
 
+	// TerminationSignals provides access to the various shutdown signals
+	// that happen during the graceful termination of the apiserver.
+	TerminationSignals *TerminationSignals
```

Comment:
why is this a pointer? Is it optional?

Reply:
it is never optional, i removed the pointer.